client/asset/dcr/dcr.go

```diff
@@ -4521,3 +4540,190 @@ func float64PtrStr(v *float64) string {
+	}
+
+	// Redemption transaction is missing from the point of view of our node!
+	// Unlikely, but possible it was redeemed by another transaction. We
+	// assume a contract past its locktime cannot make it here, so it must
+	// not be refunded. Check if the contract is still an unspent output.
+
+	swapHash, vout, err := decodeCoinID(redemption.Spends.Coin.ID())
+	if err != nil {
+		return nil, err
+	}
+
+	var spent bool
+	_, err = dcr.wallet.UnspentOutput(dcr.ctx, swapHash, vout, 0)
+	if err != nil {
+		if errors.Is(err, asset.CoinNotFoundError) {
+			spent = true
+		} else {
+			return nil, err
+		}
+	}
+
+	if spent {
+		// First find the block containing the output itself.
+		scriptAddr, err := stdaddr.NewAddressScriptHashV0(redemption.Spends.Contract, dcr.chainParams)
+		if err != nil {
+			return nil, fmt.Errorf("error encoding contract address: %w", err)
+		}
+		_, pkScript := scriptAddr.PaymentScript()
+		outFound, block, err := dcr.externalTxOutput(dcr.ctx, newOutPoint(swapHash, vout),
+			pkScript, time.Now().Add(-60*24*time.Hour)) // search up to 60 days ago
+		if err != nil {
+			return nil, err // possibly the contract is still in mempool
+		}
+		// Try to find a transaction that spends it.
+		spent, err := dcr.isOutputSpent(dcr.ctx, outFound)
+		if err != nil {
+			return nil, fmt.Errorf("error checking if contract %v:%d is spent: %w", *swapHash, vout, err)
+		}
+		if spent {
+			vin := -1
+			spendTx := outFound.spenderTx
+			for i := range spendTx.TxIn {
+				sigScript := spendTx.TxIn[i].SignatureScript
+				sigScriptLen := len(sigScript)
+				if sigScriptLen < dexdcr.SwapContractSize {
+					continue
+				}
+				// The spent contract is at the end of the signature
+				// script. Lop off the front half.
+				script := sigScript[sigScriptLen-dexdcr.SwapContractSize:]
+				_, _, _, sh, err := dexdcr.ExtractSwapDetails(script, dcr.chainParams)
+				if err != nil {
+					// This is not our script, but not necessarily
+					// a problem.
+					dcr.log.Tracef("Error encountered searching for the input that spends %v, "+
+						"extracting swap details from vin %d of %d. Probably not a problem: %v.",
+						spendTx.TxHash(), i, len(spendTx.TxIn), err)
+					continue
+				}
+				if bytes.Equal(sh[:], secretHash[:]) {
+					vin = i
+					break
+				}
+			}
+			if vin >= 0 {
+				_, height, err := dcr.wallet.GetBestBlock(dcr.ctx)
+				if err != nil {
+					return nil, err
+				}
+				hash := spendTx.TxHash()
+				confs := uint64(height - block.height)
+				if confs < requiredRedeemConfirms {
+					dcr.mempoolRedeemsMtx.Lock()
+					dcr.mempoolRedeems[secretHash] = &mempoolRedeem{txHash: hash, firstSeen: time.Now()}
+					dcr.mempoolRedeemsMtx.Unlock()
+				}
+				return &asset.ConfirmRedemptionStatus{
+					Confs:  confs,
+					Req:    requiredRedeemConfirms,
+					CoinID: toCoinID(&hash, uint32(vin)),
+				}, nil
+			}
+		}
+		dcr.log.Warnf("Contract coin %v spent by someone but not sure who.", redemption.Spends.Coin.ID())
```

Comment:
You'd also be here if `isOutputSpent` returned `spent = false`... meaning the cfilters scan just failed to fine the spender (it's actually unspent).

Reply:
Now if `isOutputSpent` return false with no error, breaking to the bottom and sending a new tx